File: futurejava/src/main/java/search/Statistics.java
package search;

public class Statistics {
    public int leafs = 0;
    public int qleafs = 0;
    public int betaCutoffs = 0;
    public int qbetaCutoffs = 0;
    public int ttHits = 0;
    public int nodes = 0;
    public int qnodes = 0;

    public int totalNodes(){
        return nodes + qnodes;
    }

    public float branchingRatio(){
        if (nodes != leafs)
            return (float)nodes / (nodes - leafs);
        return 0;
    }

    public float qBranchingRatio(){
        if (qnodes != qleafs)
            return (float)qnodes / (qnodes - qleafs);
        return 0;
    }

    public void reset(){
        leafs = 0;
        qleafs = 0;
        betaCutoffs = 0;
        qbetaCutoffs = 0;
        ttHits = 0;
        nodes = 0;
        qnodes = 0;
    }
}

File: ts-engine/src/Statistics.ts
export class Statistics {
  public leafs: number = 0;
  public qleafs: number = 0;
  public betaCutoffs: number = 0;
  public qbetaCutoffs: number = 0;
  public ttHits: number = 0;
  public nodes: number = 0;
  public qnodes: number = 0;

  public totalNodes(): number {
    return this.nodes + this.qnodes;
  }

  public branchingRatio(): number {
    if (this.nodes !== this.leafs) {
      return this.nodes / (this.nodes - this.leafs);
    }
    return 0;
  }

  public qBranchingRatio(): number {
    if (this.qnodes !== this.qleafs) {
      return this.qnodes / (this.qnodes - this.qleafs);
    }
    return 0;
  }

  public reset(): void {
    this.leafs = 0;
    this.qleafs = 0;
    this.betaCutoffs = 0;
    this.qbetaCutoffs = 0;
    this.ttHits = 0;
    this.nodes = 0;
    this.qnodes = 0;
  }
}
